app/components/application_component.rb

```diff
@@ -0,0 +1,19 @@
+class ApplicationComponent < ViewComponent::Base
+  attr_accessor :data
+  attr_writer :classes
+
+  def initialize(data: {}, classes: "")
+    self.data = data
+    self.classes = classes
+  end
+
+  # @todo this should gracefully merge left passed in data
+  def attributes(classes: "")
+    tag_builder.tag_options(data: data, class: self.classes(classes))
```

Comment:
Would

```suggestion
    tag_builder.tag_options(data: data, class: self.classes.merge(classes))
```

address the `@todo`?

(Assuming that we don't want the passed-in classes to be persisted in @classes, not sure if we do.)